test-junit5/src/main/java/io/micronaut/test/extensions/junit5/MicronautJunit5Extension.java

```diff
@@ -89,6 +95,20 @@ public void beforeAll(ExtensionContext extensionContext) throws Exception {
                 Object testInstance = extensionContext.getRequiredTestInstance();
                 applicationContext.inject(testInstance);
             }
+
+            Sql[] sqlAnnotations = testClass.getAnnotationsByType(Sql.class);
+            if (ArrayUtils.isNotEmpty(sqlAnnotations)) {
+                @SuppressWarnings("unchecked")
+                TestSqlAnnotationHandler<? super DataSource> handler = applicationContext.getBean(TestSqlAnnotationHandler.class);
+                ResourceLoader resourceLoader = applicationContext.getBean(ResourceLoader.class);
```

Comment:
Maybe the handler can support receiving multiple sql definitions, also loading datasource multiple times is not the best.

It would be to have it abstracted so R2DBC will work too.

Reply:
The idea would be to include the datasource type in the `@Sql`, by default it would be `DataSource.class`.

Reply:
So, after looking, R2DBC doesn't have a DataSource?

I've refactored to get the named datasource and unwrap it if required...

Not sure how to handle R2DBC though...any pointers?